band_steering.c

```diff
@@ -24,25 +24,41 @@ void usteer_band_steering_sta_update(struct sta_info *si)
 		si->band_steering.below_snr = true;
 }
 
-bool usteer_band_steering_is_target(struct usteer_local_node *ln, struct usteer_node *node)
+static bool band_steering_is_target(struct usteer_node *cur, struct usteer_node *new)
 {
-	if (&ln->node == node)
+	if (cur == new)
 		return false;
 
-	if (strcmp(ln->node.ssid, node->ssid))
+	if (strcmp(cur->ssid, new->ssid))
 		return false;
 
-	if (node->freq < 4000)
+	if (new->freq < 4000)
 		return false;
 
-	if (!usteer_policy_node_below_max_assoc(node))
+	if (!usteer_policy_node_below_max_assoc(new))
 		return false;
 	
 	/* ToDo: Skip nodes with active load-kick */
 	
 	return true;
- }
+}
+
+bool usteer_band_steering_is_target(struct usteer_local_node *ln, struct usteer_node *node)
```

Comment:
I'm not entirely sure the rationale for `usteer_band_steering_is_target` to accept a local node, since it only access the fields within `struct usteer_node`.